packages/node_modules/@node-red/editor-client/src/js/ui/view.js

```diff
@@ -3505,491 +3505,464 @@ RED.view = (function() {
+                statusRect.setAttribute("height",9);
+                statusRect.setAttribute("rx",2);
+                statusRect.setAttribute("ry",2);
+                statusRect.setAttribute("stroke-width","3");
+                statusEl.appendChild(statusRect);
+                node[0][0].__statusShape__ = statusRect;
+
+                var statusLabel = document.createElementNS("http://www.w3.org/2000/svg","text");
+                statusLabel.setAttribute("class","red-ui-flow-node-status-label");
+                statusLabel.setAttribute("x",20);
+                statusLabel.setAttribute("y",10);
+                statusEl.appendChild(statusLabel);
+                node[0][0].__statusLabel__ = statusLabel;
+
+                nodeContents.appendChild(statusEl);
+
+
+                var changeBadgeG = document.createElementNS("http://www.w3.org/2000/svg","g");
+                changeBadgeG.setAttribute("class","red-ui-flow-node-changed hide");
+                changeBadgeG.setAttribute("transform","translate(20, -2)");
+                node[0][0].__changeBadge__ = changeBadgeG;
+                var changeBadge = document.createElementNS("http://www.w3.org/2000/svg","circle");
+                changeBadge.setAttribute("r",5);
+                changeBadgeG.appendChild(changeBadge);
+                nodeContents.appendChild(changeBadgeG);
+
+
+                var errorBadgeG = document.createElementNS("http://www.w3.org/2000/svg","g");
+                errorBadgeG.setAttribute("class","red-ui-flow-node-error hide");
+                errorBadgeG.setAttribute("transform","translate(0, -2)");
+                node[0][0].__errorBadge__ = errorBadgeG;
+                var errorBadge = document.createElementNS("http://www.w3.org/2000/svg","path");
+                errorBadge.setAttribute("d","M -5,4 l 10,0 -5,-8 z");
+                errorBadgeG.appendChild(errorBadge);
+                errorBadge.__data__ = d;
+                errorBadge.addEventListener("mouseenter", errorBadgeMouseEnter);
+                errorBadge.addEventListener("mouseleave", errorBadgeMouseLeave);
+                nodeContents.appendChild(errorBadgeG);
+
+                node[0][0].appendChild(nodeContents);
+            });
+            node.each(function(d,i) {
+                if (d.dirty) {
                     var isLink = (d.type === "link in" || d.type === "link out")
                     var hideLabel = d.hasOwnProperty('l')?!d.l : isLink;
-                    node.attr("id",d.id);
-                    var labelWidth = calculateTextWidth(RED.utils.getNodeLabel(d), "red-ui-flow-node-label") + 50;
-                    if (d.resize || d.w === undefined) {
+                    dirtyNodes[d.id] = d;
+                    //if (d.x < -50) deleteSelection();  // Delete nodes if dragged back to palette
+
+                    var label = RED.utils.getNodeLabel(d, d.type);
+                    var labelWidth = calculateTextWidth(label, "red-ui-flow-node-label") + 50;
```

Comment:
`labelWidth` is only used inside the `if (d.resize)` block below - so this should be moved inside that if block.